src/main/java/rocks/inspectit/gepard/agent/resolver/scope/ScopeResolver.java

```diff
@@ -0,0 +1,133 @@
+package rocks.inspectit.gepard.agent.resolver.scope;
+
+import java.util.Collection;
+import java.util.List;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+import rocks.inspectit.gepard.agent.internal.configuration.model.instrumentation.InstrumentationConfiguration;
+import rocks.inspectit.gepard.agent.internal.configuration.model.instrumentation.Scope;
+import rocks.inspectit.gepard.agent.resolver.ConfigurationHolder;
+import rocks.inspectit.gepard.agent.resolver.matcher.CustomElementMatchers;
+import rocks.inspectit.gepard.agent.resolver.matcher.MatcherChainBuilder;
+
+/**
+ * This class is used to resolve the {@link Scope} based on the {@link Scope} List, contained in the
+ * {@link InstrumentationConfiguration}.
+ */
+public class ScopeResolver {
+
+  ConfigurationHolder holder;
+
+  public ScopeResolver(ConfigurationHolder holder) {
+    this.holder = holder;
+  }
+
+  /**
+   * Checks, if the provided class name requires instrumentation.
+   *
+   * @param fullyQualifiedName the full name of the class
+   * @return true, if the provided class should be instrumented
+   */
+  public boolean shouldInstrument(String fullyQualifiedName) {
+
+    List<Scope> scopes = getScopes();
+
+    return !shouldIgnore(fullyQualifiedName)
+        && scopes.stream()
+            .anyMatch(scope -> scope.getFqn().equals(fullyQualifiedName) && scope.isEnabled());
+  }
+
+  /**
+   * Builds a matcher for the methods of the provided type.
+   *
+   * @param typeDescription the type to build the method matcher for
+   * @return the matcher for the methods
+   */
+  public ElementMatcher.Junction<MethodDescription> buildMethodMatcher(
+      TypeDescription typeDescription) {
+
+    List<Scope> scopes = getAllMatchingScopes(typeDescription.getName());
+
+    if (containsWholeClassScope(scopes)) {
+      return ElementMatchers.isMethod();
+    }
+
+    List<String> methodNames = collectMethodNames(scopes);
+
+    return buildMatcherForMethods(methodNames);
+  }
+
+  /**
+   * Checks if the provided scope contains method definitions.
+   *
+   * @param scope the scope to check
+   * @return true if the scope contains method definitions
+   */
+  private boolean isWholeClassScope(Scope scope) {
+    return scope.getMethods() == null || scope.getMethods().isEmpty();
+  }
+
+  /**
+   * Checks if the provided list of scopes contains at least one whole class scope.
+   *
+   * @param scopes the list of scopes to check
+   * @return true if the list of scopes contains at least one whole class scope
+   */
+  private boolean containsWholeClassScope(List<Scope> scopes) {
```

Comment:
I'm not happy with the wording. For the whole class scope, the constructor is still missing.
Can we name it something like `containsAllMethods()`.